libs/dextool/source/dextool/compilation_db/package.d

```diff
@@ -1220,25 +1254,54 @@ unittest {
 unittest {
     auto cmd = toCompileCommand("/home", "file.cpp", [
-            "-I", "dir with spaces/src"
+            "-I", `"dir with spaces"`, "-I", `\"dir with spaces\"`
             ], AbsoluteCompileDbDirectory("/home"), null);
     auto pargs = cmd.get.parseFlag(defaultCompilerFilter, Compiler.init);
-    pargs.cflags.shouldEqual(["-I", "/home/dir with spaces/src"]);
-    pargs.includes.shouldEqual(["/home/dir with spaces/src"]);
+    pargs.cflags.shouldEqual([
+            "-I", "/home/dir with spaces", "-I", "/home/dir with spaces"
+            ]);
+    pargs.includes.shouldEqual([
+            "/home/dir with spaces", "/home/dir with spaces"
+            ]);
 }
 
-@("shall handle path with spaces, both as separate string and combined")
+@("shall handle path with spaces, both as separate string and combined with backslash")
 unittest {
     auto cmd = toCompileCommand("/project", "file.cpp", [
-            "-I", "separate dir/with space", "-Icombined dir/with space/src"
+            "-I", `"separate dir/with space"`, "-I", `\"separate dir/with space\"`,
+            `-I"combined dir/with space"`, `-I\"combined dir/with space\"`,
             ], AbsoluteCompileDbDirectory("/project"), null);
     auto pargs = cmd.get.parseFlag(defaultCompilerFilter, Compiler.init);
     pargs.cflags.shouldEqual([
             "-I", "/project/separate dir/with space", "-I",
-            "/project/combined dir/with space/src"
+            "/project/separate dir/with space", "-I",
+            "/project/combined dir/with space", "-I",
+            "/project/combined dir/with space"
+            ]);
+    pargs.includes.shouldEqual([
+            "/project/separate dir/with space", "/project/separate dir/with space",
+            "/project/combined dir/with space", "/project/combined dir/with space"
+            ]);
+}
+
+@("shall handle path with consecutive spaces")
+unittest {
+    auto cmd = toCompileCommand("/project", "file.cpp", [
+            `-I"one`, `space/lots`, `of     space"`, `-I\"one`, `space/lots`,
+            `of     space\"`, `-I`, `"one`, `space/lots`, `of     space"`,
+            `-I`, `\"one`, `space/lots`, `of     space\"`,
+            ], AbsoluteCompileDbDirectory("/project"), null);
+    auto pargs = cmd.get.parseFlag(defaultCompilerFilter, Compiler.init);
+    pargs.cflags.shouldEqual([
+            "-I", "/project/one space/lots of     space", "-I",
+            "/project/one space/lots of     space", "-I",
+            "/project/one space/lots of     space", "-I",
+            "/project/one space/lots of     space",
             ]);
     pargs.includes.shouldEqual([
-            "/project/separate dir/with space",
-            "/project/combined dir/with space/src"
+            "/project/one space/lots of     space",
+            "/project/one space/lots of     space",
+            "/project/one space/lots of     space",
+            "/project/one space/lots of     space"
             ]);
```

Comment:
These test cases are meant to test the example provided in #938. When printing the args in the filterPair-function, the following is the output:
```
-I"/home/niklas/Projects/mutant
ninjas/test
project/tests/../src"
```
where newlines are spaces in the path. I am not sure, however, that two of the cases can ever happen. For example:
```
`-I`, `\"one`, `space/lots`, `of     space\"`
```
Anyhow, the test is there, and the implementation handles it. Should we keep it "in case of"?

Reply:
Correct, the case above can never happen.
it is either:
```d
[`-I`, `\\"one space/lots of     spaces\\"`]
```
or
```d
[`-I\\"one space/lots of     spaces\\"`]
```